### ISSUE TITLE
Can not generate right schema with @ArraySchema and oneOf

### ISSUE THREAD

**Issue by lociko** [swagger-api/swagger-core#2616]
| Q                                 | A
| --------------------------------- | -------
| Bug or feature request?           | bug
| Which Swagger-Core version?| 2.0.0-rc1 
| Which Java version?               | java-8
| Which JAX-RS framework & version? | jersey-2

I go into troubles trying to annotate my models. The problem is that `@ArraySchema` ignores `oneOf` property in `@Schema`.

In the example, belove is shown my DTO classes. 
```
@Schema(name = "testObject")
public class TestObject {

    private List<AbstractObject> objects;
    @ArraySchema(schema = @Schema(
            name = "objects",
            oneOf = {
                    AObject.class,
                    BObject.class
            }
    ))
    public List<AbstractObject> getObjects() {return objects;}

    public void setObjects(List<AbstractObject> objects) {this.objects = objects;}
}

abstract class AbstractObject {}

@Schema(name = "AObject")
class AObject extends AbstractObject{
    private String name;

    public String getName() {return name;}

    public void setName(String name) {this.name = name;}
}

@Schema(name = "BObject")
class BObject extends AbstractObject{
    private Integer number;

    public Integer getNumber() {return number;}

    public void setNumber(Integer number) {this.number = number;}
}
```
I have a testObject which have one property which is an array of AbstractObject. If a wan to generate an OpenAPI schema I get a list of component like this:
```
components:
  schemas:
    AbstractObject:
      type: object
    testObject:
      type: object
      properties:
        objects:
          type: array
          items:
            type: array
            items:
              $ref: '#/components/schemas/AbstractObject'
```

In that schema, I do not see any of my classes and relations between them. 
**I expect to see something like this:**
```
schemas:
    testObject:
      type: object
      properties:
        object:
          type: array
          items:
            type: object
            oneOf:
            - $ref: '#/components/schemas/AObject'
            - $ref: '#/components/schemas/BObject'
    AObject:
      type: object
      properties:
        name:
          type: string
    BObject:
      type: object
      properties:
        number:
          type: integer
          format: int32
```

**Comment by lociko** [comment]
Any updates?

**Comment by frantuma** [comment]
updated behaviour has been implemented in #2692, available in latest `2.0.0-SNAPSHOT` (on sonatype) and upcoming release. The outcome is similar to expected one in comment above:

```
AObject:
  type: object
  properties:
    name:
      type: string
AbstractObject:
  type: object
  oneOf:
  - $ref: '#/components/schemas/AObject'
  - $ref: '#/components/schemas/BObject'
BObject:
  type: object
  properties:
    number:
      type: integer
      format: int32
testObject:
  type: object
  properties:
    objects:
      type: array
      items:
        $ref: '#/components/schemas/AbstractObject'
```

Closing ticket, please reopen if you're still experiencing issues.
